src/System.Buffers.Primitives/System/Buffers/ReaderWriter/BufferReader.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Sequences;
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers
+{
+    public class BufferReader
+    {
+        public static BufferReader<TSequence> Create<TSequence>(TSequence buffer) where TSequence : ISequence<ReadOnlyMemory<byte>>
+        {
+            return new BufferReader<TSequence>(buffer);
+        }
+    }
+
+    public ref partial struct BufferReader<TSequence> where TSequence : ISequence<ReadOnlyMemory<byte>>
+    {
+        private TSequence _sequence;
```

Comment:
Expose this as public, please. Having ability to write extension methods for `BufferReader<ROB>` and use `ROB` is very nice.

Reply:
Let's do it separately. I worry that consumers will need more than the sequence to make it work.

Reply:
Let's just do it, it's super easy, and I was doing in other PR that I had to close.